### DISCUSSION TITLE
Tanstack router with electron-vite-react

### DISCUSSION THREAD

**Discussion by Aliwahid17** [TanStack/router#835] [Q&A]
I'm working on an electron project but in dev everything is working fine but the build is not working. 

Please 🙏 let me know how to fix that

**Comment by mathias-berg** [comment]
Instead of `createBrowserHistory` you should use `createHashHistory`.

https://tanstack.com/router/v1/docs/framework/react/guide/history-types#hash-routing

**Comment by theorib** [comment]
All you have to do is create a `memoryHistory` and pass it to `createRouter` when you are first initialising it. You don't have to use `createHashHistory` (at least that's how I'm using it and I still haven't stumbled into any issues):

```typescript
// Create a memory history instance to initialize the router so it doesn't break when compiled:
const memoryHistory = createMemoryHistory({
  initialEntries: ['/'] // Pass your initial url
})
// Create a new router instance
const router = createRouter({ routeTree, history: memoryHistory })
```

Check [this page](https://tanstack.com/router/v1/docs/framework/react/guide/history-types) on the docs for more info. [This](https://discord.com/channels/719702312431386674/1204162137098944623) is where I found the solution on Tanstack's discord channel.

**Comment by neisukee** [comment]
For my project, `createHashHistory` worked better.

Since I allow users to reload and navigate forward and back in my Electron app, `createMemoryHistory` didn't play well with the browser's history functionality.

Using `createHashHistory` gives users the familiar browser navigation experience they expect.
